geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterQueryPlanner.scala

```diff
@@ -0,0 +1,146 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.locationtech.geomesa.raster.data
+
+import com.google.common.collect.ImmutableSetMultimap
+import com.typesafe.scalalogging.slf4j.Logging
+import com.vividsolutions.jts.geom.Geometry
+import org.apache.accumulo.core.client.IteratorSetting
+import org.apache.accumulo.core.data.{Range => ARange}
+import org.apache.hadoop.io.Text
+import org.geotools.factory.CommonFactoryFinder
+import org.geotools.filter.text.ecql.ECQL
+import org.locationtech.geomesa.accumulo._
+import org.locationtech.geomesa.accumulo.index.{IndexFilterHelpers, QueryPlan, _}
+import org.locationtech.geomesa.accumulo.iterators._
+import org.locationtech.geomesa.accumulo.process.knn.TouchingGeoHashes
+import org.locationtech.geomesa.raster.iterators.RasterFilteringIterator
+import org.locationtech.geomesa.raster.{lexiEncodeDoubleToString, rasterSft, rasterSftName}
+import org.locationtech.geomesa.utils.geohash.{BoundingBox, GeohashUtils}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.filter.Filter
+
+import scala.collection.JavaConversions._
+import scala.util.Try
+
+// TODO: Constructor needs info to create Row Formatter
+// TODO: Consider adding resolutions + extent info  https://geomesa.atlassian.net/browse/GEOMESA-645
+class AccumuloRasterQueryPlanner extends Logging with IndexFilterHelpers {
+
+  def modifyHashRange(hash: String, expectedLen: Int, res: String): ARange = expectedLen match {
+    case 0                                     => new ARange(new Text(s"$res~"))
+    case lucky if expectedLen == hash.length   => new ARange(new Text(s"$res~$hash"))
+    case shorten if expectedLen < hash.length  => new ARange(new Text(s"$res~${hash.substring(0, expectedLen)}"))
+    case lengthen if expectedLen > hash.length => new ARange(new Text(s"$res~$hash"), new Text(s"$res~$hash~"))
+  }
+
+  def getQueryPlan(rq: RasterQuery, resAndGeoHashMap: ImmutableSetMultimap[Double, Int]): Option[QueryPlan] = {
+    val availableResolutions = resAndGeoHashMap.keySet().toList.sorted
+
+    // Step 1. Pick resolution
+
+    val selectedRes: Double = selectResolution(rq.resolution, availableResolutions)
+    val res = lexiEncodeDoubleToString(selectedRes)
+
+    // Step 2. Pick GeoHashLength
+    val GeoHashLenList = resAndGeoHashMap.get(selectedRes).toList
+    val expectedGeoHashLen = if (GeoHashLenList.isEmpty) {
```

Comment:
[picky][style] You might turn this into a one-liner and dispense with the braces.